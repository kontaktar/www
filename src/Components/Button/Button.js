import React from "react";
import styles from "./Button.module.scss";

class Button extends React.Component {
  render() {
    return (
      <div className={styles.button}>
        <p>Hello World!</p>
      </div>
    );
  }
}

export default Button;
