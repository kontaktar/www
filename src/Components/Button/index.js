import React from "react";
import styles from "./index.scss";

class Button extends React.Component {
  render() {
    return (
      <div className={styles.button}>
        <h2>Button component</h2>
      </div>
    );
  }
}

export default Button;
