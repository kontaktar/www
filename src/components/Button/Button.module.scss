@import "../../styles/config";

.button {
  height: 50px;
  width: 202px;
  cursor: pointer;

  background-color: $primary-color;
  border-color: $primary-color;
  border-style: solid;
  color: $color-white;

  font-size: 18px;
  font-family: "Berlingske Sans Bold";
  font-weight: 600;
  line-height: 144%;
  text-align: center;
  letter-spacing: 0.01em;
  font-style: bold;

  &:hover,
  &:active,
  &:focus {
    background-color: $color-white;
    border: 1px solid $primary-color;
    color: $primary-color;

    transition: all 0.5s ease;
  }

  &:disabled {
    background-color: $color-white;
    border: 1px solid $color-brown;
    color: $color-brown;

    transition: all 0.5s ease;
  }
}

.edit {
  border-radius: 5px;
  border: none;
  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: space-around;
  width: auto;
  box-shadow: 0px 4px 30px rgba(0, 0, 0, 0.1);
  p {
    padding: 0 24px;
    color: white;
    font-size: 16px;
    font-family: Berlingske Sans;
  }
  .icon {
    padding-right: 24px;
  }
}
.edit.publish {
  background-color: $color-green;
}
.edit.save {
  background-color: $color-orange;
}

.navigation:before,
.navigation:after {
  content: "";
  display: table;
  clear: both;
}
.navigation {
  font-size: 16px;
  height: 24px;
  background-color: white;
  display: flex;
  align-items: center;
  justify-items: space-around;
  border: none;
  flex-direction: row;
  width: 100%;

  p {
    color: $color-grey-light;
    padding-top: 4px;
    line-height: normal;
    font-family: "Berlingske Sans Bold";
    margin-left: 0px;
    transition: margin 0.3s ease;

    white-space: nowrap;
  }

  box-shadow: inset 0 0 0 0 $primary-color;

  &:hover,
  &:focus,
  &:active {
    border: none;
    p {
      color: black;
    }

    box-shadow: inset 4px 0 0 0 $primary-color;
    transition: box-shadow ease-in-out 0.3s;
  }
  .navigation_icon {
    margin: 0 30px;
    transition: margin 0.4s;
  }

  svg {
    display: block;
    margin: auto;
  }
}

.compact {
  p {
    visibility: hidden;
    opacity: 0;
    transition: visibility 0.3s linear, opacity 0.3s linear;
    margin-left: 10px;
    transition: margin 0.3s linear;
  }
  .navigation_icon {
    margin: 0 15px;
  }
}

.pill {
  border-radius: 24px;
}
.inverted {
  background-color: $color-white;
  color: $primary-color;
  border-color: $primary-color;
  border-image: none;
  &:hover,
  &:active,
  &:focus {
    cursor: pointer;
    background-color: $primary-color;
    color: $color-white;
  }
}
.borderless {
  border: none;
  color: $paragraph-color;
  background-color: $color-white;
  width: 100px;
  margin: 0 10px;
  &:hover,
  &:active,
  &:focus {
    cursor: pointer;
    border: none;
  }
}
.search {
  position: absolute;
  border: none;
  top: 12px;
  right: 1px;
  box-sizing: content-box;
  background-color: white;
  width: 40px;
  height: 32px;

  &:hover,
  &:active,
  &:focus {
    border: none;
    path {
      fill: black;
      fill-opacity: 1;
    }
  }
}

.rectangle {
  height: 67px;
  width: 67px;
}

.carousel_navi {
  .back {
    transform: rotate(90deg);
    * {
      fill: $primary-color;
    }
  }
  .next {
    transform: rotate(270deg);
    * {
      fill: white;
    }
  }
  &:hover,
  &:active,
  &:focus {
    .back * {
      fill: white;
    }
    .next * {
      fill: $primary-color;
    }
  }
}

.hamburger {
  width: 50px;
  height: 50px;
  padding: 8px;
  background: white;
  border: 1px solid white;

  &:hover,
  &:active {
    background-color: $color-white;
    border: 1px solid white;
    color: $primary-color;

    *:first-child {
      opacity: 0;
      transition: opacity 0.3s ease;
    }
    *:nth-child(2) {
      opacity: 0;
      transition: opacity 0.6s ease;
    }
  }

  * {
    height: 6px;
    margin-top: 7px;
    background: $primary-color;
  }
  *:first-child {
    margin-top: 0;
  }
}
