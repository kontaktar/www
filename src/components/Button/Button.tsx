import React, { ReactChild, ReactElement } from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import cx from "classnames";
import { Icon } from "components";
import Link from "components/LinkWrap";
import styles from "./Button.module.scss";
import colors from "styles/colors.module.scss";

type Props = {
  children: ReactChild;
  className?: string;
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<Element, MouseEvent>) => void;
  modifier?: Array<string>; // TODO: onez["", "pill", "inverted", "borderless", "rectangle", "raw"]
  isLoading?: boolean;
  type?: string;
  [x: string]: any;
};

interface StaticComponents {
  Navigation?: Props;
  CarouselNavi?: Props;
  Edit?: Props;
  Search?: Props;
  Hamburger?: Props;
}
const Button = ({
  children,
  className = "",
  disabled,
  onClick,
  modifier,
  isLoading,
  type,
  name,
  ...props
}: Props & StaticComponents): ReactElement => {
  return (
    <button
      className={`${className} ${styles.button} ${modifier?.map(
        (m: string) => ` ${styles[m]} `
      )}`}
      // eslint-disable-next-line react/destructuring-assignment
      // data-test={props?.["data-test"] || `${name}Button`}
      id={name}
      name={name}
      disabled={disabled}
      onClick={onClick}
      type={type === "submit" ? "submit" : "button"}
      {...props}
      data-test={props?.["data-test"] || `${name}Button`}
    >
      {isLoading ? <CircularProgress /> : <>{children}</>}
    </button>
  );
};

const ButtonNavigation = ({
  compact = false,
  title,
  url,
  icon,
  onClick,
  ...props
}) => {
  return (
    <Link href={url} onClick={onClick}>
      <Button
        className={`${styles.navigation} ${compact && styles.compact}`}
        {...props}
      >
        <>
          <Icon
            color={colors.greyLight}
            className={styles.navigation_icon}
            name={icon}
          />
          <p>{title}</p>
        </>
      </Button>
    </Link>
  );
};

const ButtonCarouselNavi = ({ direction, className, ...props }) => {
  const modifers =
    direction === "next" ? ["rectangle"] : ["rectangle", "inverted"];
  return (
    <Button
      className={`${styles.carousel_navi} ${className}`}
      modifier={modifers}
      {...props}
    >
      <Icon
        className={`${direction === "next" ? styles.next : styles.back}`}
        name="arrow"
      />
    </Button>
  );
};

const ButtonEdit = ({ type, className, ...props }) => {
  return (
    <Button
      data-test={
        type === "save" ? "saveUserInfoButton" : "publishUserInfoButton"
      }
      type="submit"
      className={cx(className, styles.edit, styles[type])}
      {...props}
    >
      <>
        <p>{type === "save" ? "Vista" : "Birta"}</p>
        <Icon
          className={styles.icon}
          name={type === "save" ? "save" : "publish"}
        />
      </>
    </Button>
  );
};
const ButtonDelete = ({ className, ...props }) => {
  return (
    <Button
      type="button"
      data-test="deleteUser"
      className={cx(className, styles.delete)}
      {...props}
    >
      EY√êA NOTANDA
    </Button>
  );
};

const ButtonSearch = () => {
  return (
    <Button>
      <Icon className={styles.icon} name="search-large" />
    </Button>
  );
};

const ButtonHamburger = (props) => {
  return (
    <Button className={styles.hamburger} {...props}>
      <p />
      <p />
      <p />
    </Button>
  );
};

Button.Navigation = ButtonNavigation;
Button.CarouselNavi = ButtonCarouselNavi;
Button.Edit = ButtonEdit;
Button.Delete = ButtonDelete;
Button.Search = ButtonSearch;
Button.Hamburger = ButtonHamburger;

export default Button;
