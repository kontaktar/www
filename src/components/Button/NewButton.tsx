import React, { ReactChild, ReactElement } from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import styles from "./Button.module.scss";

type Props = {
  children: ReactChild;
  className?: string;
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<Element, MouseEvent>) => void;
  modifier?: Array<string>;
  isLoading?: boolean;
  type?: string;
  [x: string]: any;
};
const Button = ({
  children,
  className = "",
  disabled,
  onClick,
  modifier,
  isLoading,
  type,
  name,
  ...props
}: Props): ReactElement => {
  return (
    <button
      className={`${className} ${styles.button} ${modifier?.map(
        (m: string) => ` ${styles[m]} `
      )}`}
      // eslint-disable-next-line react/destructuring-assignment
      // data-test={props?.["data-test"] || `${name}Button`}
      id={name}
      name={name}
      disabled={disabled}
      onClick={onClick}
      type={type === "submit" ? "submit" : "button"}
      {...props}
      data-test={props?.["data-test"] || `${name}Button`}
    >
      {isLoading ? <CircularProgress /> : <>{children}</>}
    </button>
  );
};
export default Button;
