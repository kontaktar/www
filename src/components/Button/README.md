```js
import { <%= name %> } from "components";
```

Replace this document with real info

<!-- PROPS -->
<!-- A propsTable will be rendered here in Storybook -->
