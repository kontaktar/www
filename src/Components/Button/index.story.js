import React from "react";
import { storiesOf } from "@storybook/react";
import { withReadme, withDocs, doc } from "storybook-readme";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import Button from ".";
import test from "./test.md";

storiesOf("Button", module)
  .addDecorator(withKnobs)
  .addDecorator(withDocs(test))
  .add("inital story", () => <Button />);

storiesOf("Button/Rounded", module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(test))
  .add("inital sstory", () => <Button />);
