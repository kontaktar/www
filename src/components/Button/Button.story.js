/* eslint-disable unicorn/prevent-abbreviations */
import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Button from "./Button";
import ButtonReadme from "./README.md";

storiesOf("Components", module)
  .addParameters({
    jest: ["Button.test"],
    readme: {
      sidebar: ButtonReadme,
      includePropTables: [Button]
    }
  })
  .add("Button", () => (
    <div style={{ padding: "20px" }}>
      <p>No modifier</p>
      <Button
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
      >
        {text("Label", "This is a button")}
      </Button>
      <p>Pill modifier</p>
      <Button
        id="button2"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
        modifier={["pill"]}
      >
        {text("Label", "This is a button")}
      </Button>
      <p>Inverted modifier</p>
      <Button
        id="button2"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
        modifier={["inverted"]}
      >
        {text("Label", "This is a button")}
      </Button>
      <p>Disabled</p>
      <Button id="button2" disabled onClick={action("button-clicked")}>
        {text("Label", "This is a button")}
      </Button>
      <Button.Navigation />
      <Button.Navigation compact />
      <p>Rectangle modifier</p>
      <Button
        id="button2"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
        modifier={["rectangle"]}
      />
      <p>Rectangle and inverted modifier</p>
      <Button
        id="button2"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
        modifier={["rectangle", "inverted"]}
      />
      <p>CarouseleNavi next</p>
      <Button.CarouselNavi
        id="button2"
        direction="next"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
      />
      <p>CarouseleNavi back</p>
      <Button.CarouselNavi
        id="button2"
        direction="back"
        disabled={boolean("Disabled", false)}
        onClick={action("button-clicked")}
      />
      <Button.Edit type="save" />
      <Button.Edit type="publish" />
    </div>
  ));
