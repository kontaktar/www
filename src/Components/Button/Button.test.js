import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Button from "./Button.js";

describe("With Enzyme", () => {
  it('Component shows "Hello world!"', () => {
    const app = shallow(<Button />);

    expect(app.find("p").text()).toEqual("Hello World!");
  });
});

describe("With Snapshot Testing", () => {
  it('App shows "Hello world!"', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
