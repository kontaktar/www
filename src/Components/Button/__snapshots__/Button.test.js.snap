// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div
  className="button"
>
  <p>
    Hello World!
  </p>
</div>
`;
